build/flags.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+set -e
+
+if [ ! -f "build/env.sh" ]; then
+    echo "$0 must be run from the root of the repository."
+    exit 2
+fi
+
+GO_ETHEREUM_FLAGS=""
+
+# set gitCommit when running from a Git checkout.
+if [ -f ".git/HEAD" ]; then
+    GO_ETHEREUM_FLAGS=$GO_ETHEREUM_FLAGS"-ldflags '-X main.gitCommit $(git rev-parse HEAD)'"
+fi
+
+: ${GO_OPENCL:="false"}
```

Comment:
I think it could be as simple as the following snippet:

```shell
if [ -f ".git/HEAD" ]; then 
    echo "-ldflags '-X main.gitCommit $(git rev-parse HEAD)'"
fi

if [ ! -z "$GO_OPENCL" ]; then
   echo " -tags opencl"
fi
```